addons/website/static/src/components/dialog/add_page_dialog.js

```diff
@@ -445,20 +450,30 @@ export class AddPageDialog extends Component {
         this.lastTabName = name;
     }
 
-    async addPage(sectionsArch, name) {
+    async addPage(sectionsArch, name, templateId) {
         if (this.props.forcedURL) {
             // We also skip the possibility to choose to add in menu in that
             // case (e.g. in creation from 404 page button). The user can still
             // create its menu afterwards if needed.
             await this.createPage(sectionsArch, this.props.forcedURL, false, this.props.pageTitle);
         } else {
-            this.dialogs.add(AddPageConfirmDialog, {
-                createPage: (...args) => this.createPage(sectionsArch, ...args),
+            // Hook for extensions to customize the dialog class
+            const DialogClass = this.getConfirmDialogInstance();
+            this.dialogs.add(DialogClass, {
+                createPage: (...args) => this.createPage(...args),
                 name: name || this.lastTabName,
+                sectionsArch: sectionsArch || "",
+                templateId: templateId || "",
             });
         }
     }
 
+    getConfirmDialogInstance() {
+        // This is a hook for extensions to customize the dialog class
+        // Overridden in enterprise/ai_website to use AddPageAIConfirmDialog
+        return AddPageConfirmDialog;
+    }
```

Comment:
See comment on the enterprise PR for more details: you should not change the whole dialog class in fact (even replacing by one that extends our base one), you should path the base one in place.